src/cls.ts

```diff
@@ -16,47 +16,37 @@
 
 'use strict';
 
-var semver = require('semver');
-var cls = semver.satisfies(process.version, '>=8') &&
-          process.env.GCLOUD_TRACE_NEW_CONTEXT ?
-            require('./cls-ah'/*.js*/) : require('continuation-local-storage');
+import * as semver from 'semver';
+import * as CLS from 'continuation-local-storage';
 
-/** @const {string} */
-var TRACE_NAMESPACE = 'com.google.cloud.trace';
+const cls: typeof CLS = semver.satisfies(process.version, '>=8') &&
+  process.env.GCLOUD_TRACE_NEW_CONTEXT ? require('./cls-ah')
+                                       : require('continuation-local-storage');
 
-function createNamespace() {
+const TRACE_NAMESPACE = 'com.google.cloud.trace';
+
+export function createNamespace(): CLS.Namespace {
   return cls.createNamespace(TRACE_NAMESPACE);
 }
 
-function destroyNamespace() {
+export function destroyNamespace(): void {
   cls.destroyNamespace(TRACE_NAMESPACE);
 }
 
-function getNamespace() {
+export function getNamespace(): CLS.Namespace {
   return cls.getNamespace(TRACE_NAMESPACE);
 }
 
-module.exports = {
-  createNamespace: createNamespace,
-
-  destroyNamespace: destroyNamespace,
-
-  getNamespace: getNamespace,
-
-  getRootContext: function getRootContext() {
-    // First getNamespace check is necessary in case any
-    // patched closures escaped before the agent was stopped and the
-    // namespace was destroyed.
-    if (getNamespace() && getNamespace().get('root')) {
-      return getNamespace().get('root');
-    }
-    return null;
-  },
-
-  setRootContext: function setRootContext(rootContext) {
-    getNamespace().set('root', rootContext);
+export function getRootContext(): any {
+  // First getNamespace check is necessary in case any
+  // patched closures escaped before the agent was stopped and the
+  // namespace was destroyed.
+  if (getNamespace() && getNamespace().get('root')) {
+    return getNamespace().get('root');
   }
-};
-
+  return null;
+}
 
-export default {};
+export function setRootContext(rootContext: any): void {
```

Comment:
Do we ever use anything other than spans?

Reply:
Same as in `getRootContext`